### ISSUE TITLE
PGO build: Errors about missing src/pgo-timestamp.h

### ISSUE THREAD

**Issue by dgelessus** [SWI-Prolog/swipl-devel#860]
I'm trying to do a PGO build of 05a40addb0c58302d60ed4fd27ccc332731f45ea on macOS 10.14. Currently I'm using Clang - I'll try with GCC later, but I don't think it will change anything about this error. The instrumented build and profiling step completes successfully, but the main build then fails with these errors:

```
[ 71%] Generating ../PGO-data/swipl.profdata
[ 71%] Generating pgo-timestamp.h
[ 71%] Built target pgo_data
Scanning dependencies of target swiplobjs
[ 71%] Building C object src/CMakeFiles/swiplobjs.dir/pl-atom.c.o
[ 71%] Building C object src/CMakeFiles/swiplobjs.dir/pl-arith.c.o
[ 71%] Building C object src/CMakeFiles/swiplobjs.dir/pl-wam.c.o
[ 71%] Building C object src/CMakeFiles/swiplobjs.dir/pl-bag.c.o
[ 71%] Building C object src/CMakeFiles/swiplobjs.dir/pl-error.c.o
<built-in>:1:10: fatal error: 'src/pgo-timestamp.h' file not found
#include "src/pgo-timestamp.h"
         ^~~~~~~~~~~~~~~~~~~~~
<built-in>:1:10: fatal error: 'src/pgo-timestamp.h' file not found
#include "src/pgo-timestamp.h"
         ^~~~~~~~~~~~~~~~~~~~~
<built-in>:1:10: <built-in>:1:10: fatal error: 'src/pgo-timestamp.h' file not found
#include "src/pgo-timestamp.h"
         ^~~~~~~~~~~~~~~~~~~~~
<built-in>:1:10: fatal error: 'src/pgo-timestamp.h' file not found
#include "src/pgo-timestamp.h"
         ^~~~~~~~~~~~~~~~~~~~~
fatal error: 'src/pgo-timestamp.h' file not found
#include "src/pgo-timestamp.h"
         ^~~~~~~~~~~~~~~~~~~~~
```

The src/pgo-timestamp.h header has been generated under the build directory, but apparently the compiler isn't looking there. Here's how the compiler is being called, according to `cmake --build . --verbose` (`<swipl-devel>` is the path to where I cloned the repo):

```
[ 71%] Building C object src/CMakeFiles/swiplobjs.dir/pl-arith.c.o
cd <swipl-devel>/cmake-build/pgo/src && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc  -I<swipl-devel>/cmake-build/pgo/src -I<swipl-devel>/src -I/opt/local/include -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.14 -fPIC -fvisibility=hidden -Wall -fprofile-use=<swipl-devel>/cmake-build/pgo/PGO-data/swipl.profdata -include src/pgo-timestamp.h -o CMakeFiles/swiplobjs.dir/pl-arith.c.o -c <swipl-devel>/src/pl-arith.c
```

**Comment by JanWielemaker** [comment]
Doesn't happen with Gcc, also not on the Mac (I build the releases that way).   Surely at some point the PGO built with clang, but where PGO gives a 40% performance boost for gcc (which produces already better code to begin with), using PGO with Clang actually makes things _worse_ :cry:    We suspect Clang isn't very good at dealing with large functions.    The VM interpreter is a huge function.

Still, it is a bug ...

**Comment by JanWielemaker** [comment]
Turns out this triggers when building with `make` rather than `ninja`

**Comment by JanWielemaker** [comment]
Should be fixed with 36caa67738f157420fa27ff690f8e619057d062c.   Thanks to a tip by @dmchurch
